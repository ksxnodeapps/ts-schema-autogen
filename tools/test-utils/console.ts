import { ConsoleInstance } from 'simple-fake-console'
import { Console } from '@ts-schema-autogen/types'

export class FakeConsole implements Console.Mod {
  public readonly core = new ConsoleInstance()
  public readonly info = jest.fn(this.core.info)
  public readonly error = jest.fn(this.core.error)
}
