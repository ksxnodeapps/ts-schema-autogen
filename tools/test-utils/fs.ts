import { StringPathFileSystem } from 'simple-fake-fs'
import { FSX } from '@ts-schema-autogen/types'

export class FakeFileSystem extends StringPathFileSystem implements FSX.Mod {
  private mkfn<Xs extends any[], Y> (fn: (...xs: Xs) => Y) {
    return jest.fn(async (...xs: Xs) => fn.apply(this, xs))
  }

  public readonly stat = this.mkfn(this.statSync)
  public readonly readdir = this.mkfn(this.readdirSync)
  public readonly readFile = this.mkfn(this.readFileSync)
  public readonly outputFile = this.mkfn(this.outputFileSync)
  public readonly writeFile = this.mkfn(this.writeFileSync)
  public readonly remove = jest.fn()
}
