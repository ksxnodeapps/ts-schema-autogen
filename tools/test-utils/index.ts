export * from './console'
export * from './fs'
export * from './path'
export * from './print-result'
export * from './process'
export * from './typescript-json-schema'
