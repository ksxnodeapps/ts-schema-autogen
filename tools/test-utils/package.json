{
  "name": "@tools/test-utils",
  "version": "0.0.0",
  "private": true,
  "dependencies": {
    "simple-fake-fs": "^0.0.8",
    "simple-fake-console": "^0.1.6",
    "simple-fake-path": "^0.0.2",
    "@ts-schema-autogen/status": "file:../../packages/status",
    "@ts-schema-autogen/types": "file:../../packages/types"
  }
}
