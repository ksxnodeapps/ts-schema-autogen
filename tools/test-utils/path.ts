import { Path } from '@ts-schema-autogen/types'

export class FakePath implements Path.Mod {
  constructor (private readonly cwd: string) {}

  public readonly join = (...paths: string[]): string => {
    if (!paths.length) return '.'
    const [head, ...rest] = paths
    const tail = this.join(...rest)
    if (head === '.') return tail
    if (tail === '.') return head
    if (tail.startsWith('../')) return this.join(this.dirname(head), tail.slice('../'.length))
    const left = head.endsWith('/') ? head.slice(0, -1) : head
    const right = tail.startsWith('/') ? tail.slice(1) : tail
    return left + '/' + right
  }

  public readonly resolve = (...paths: string[]): string => {
    if (!paths.length) return '.'
    const [head, ...rest] = paths
    const tail = this.resolve(...rest)
    if (tail.startsWith('/')) return tail
    return this.join(this.cwd, head, tail)
  }

  public readonly dirname = (path: string): string => {
    if (path === '' || path === '.' || path === '/') return path
    const segments = path.split('/').slice(0, -1)
    if (!segments.length) return '.'
    return this.join(...segments)
  }

  public readonly basename = (path: string): string => {
    const segments = path.split('/')
    return segments[segments.length - 1]
  }
}
