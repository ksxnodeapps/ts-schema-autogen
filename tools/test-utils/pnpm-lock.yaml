dependencies:
  '@ts-schema-autogen/types': 'link:../../packages/types'
  simple-fake-fs: 0.0.4
lockfileVersion: 5.1
packages:
  /@types/node/13.5.0:
    dev: false
    resolution:
      integrity: sha512-Onhn+z72D2O2Pb2ql2xukJ55rglumsVo1H6Fmyi8mlU9SvKdBk/pUSUAiBY/d9bAOF7VVWajX3sths/+g6ZiAQ==
  /simple-fake-fs/0.0.4:
    dependencies:
      '@types/node': 13.5.0
      tslib: 1.10.0
    dev: false
    resolution:
      integrity: sha512-GUwEIYXI6oSJdMZkP1hK3x084z6HX6uyHMrs3erJXEzdyMIrJ7HJEjlbtMrs6zsELvgx0wr7Y9zTK61+igHC8A==
  /tslib/1.10.0:
    dev: false
    resolution:
      integrity: sha512-qOebF53frne81cf0S9B41ByenJ3/IuH8yJKngAX35CmiZySA0khhkovshKK+jGCaMnVomla7gVlIcc3EvKPbTQ==
specifiers:
  '@ts-schema-autogen/types': ../../packages/types
  simple-fake-fs: ^0.0.4
