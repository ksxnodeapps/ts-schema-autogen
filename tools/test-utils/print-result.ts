import { inspect } from 'util'
import { Success, Failure } from '@ts-schema-autogen/status'

export function printFailure (result: Failure<unknown>): string {
  let text = '\n'
  result.print((...args) => text += args.join(' ') + '\n')
  return text
}

export const printResult = (result: Success<unknown> | Failure<unknown>): string =>
  result.code ? printFailure(result) : inspect(result)
