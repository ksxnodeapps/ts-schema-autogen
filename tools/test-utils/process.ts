import { Process } from '@ts-schema-autogen/types'

export class FakeProcess implements Process.Mod {
  constructor (private readonly workDir: string) {}
  public readonly cwd = jest.fn(() => this.workDir)
}
