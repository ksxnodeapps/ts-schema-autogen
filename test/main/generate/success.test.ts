import fsTree from '../../fixtures/fs-tree/valid-config-only'
import TJS from '../../fixtures/tjs'
import { FakeConsole, FakeFileSystem, FakePath, FakeProcess } from '@tools/test-utils'

import {
  Status,
  DEFAULT_IGNORED,
  DEFAULT_PATTERN,
  cmdGenerate
} from '@ts-schema-autogen/main'

async function setup () {
  const console = new FakeConsole()
  const fsx = new FakeFileSystem('/', { wdir: fsTree })
  const path = new FakePath()
  const process = new FakeProcess('wdir')
  const tjs = new TJS()
  const result = await cmdGenerate({
    args: {
      ignored: DEFAULT_IGNORED,
      pattern: DEFAULT_PATTERN
    },
    modules: {
      console,
      fsx,
      path,
      process,
      tjs
    }
  })
  return {
    console,
    fsx,
    path,
    process,
    tjs,
    result
  }
}

it('returns Status.Success', async () => {
  const { result } = await setup()
  expect(result).toBe(Status.Success)
})

it('calls fsx.outputFile', async () => {
  const { fsx } = await setup()
  expect(fsx.outputFile.mock.calls).toMatchSnapshot()
})

it('calls console.info', async () => {
  const { console } = await setup()
  expect(console.info.mock.calls).toMatchSnapshot()
})

it('does not call console.error', async () => {
  const { console } = await setup()
  expect(console.error).not.toBeCalled()
})

it('output contents', async () => {
  const { fsx } = await setup()
  const snapshot = fsx.outputFile.mock.calls
    .map(([filename, content]) => ({ filename, content: JSON.parse(content) }))
  expect(snapshot).toMatchSnapshot()
})

it('messages', async () => {
  const { console } = await setup()
  expect(console.getString()).toMatchSnapshot()
})
