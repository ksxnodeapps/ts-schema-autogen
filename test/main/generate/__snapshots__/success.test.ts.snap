// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls console.info 1`] = `
Array [
  Array [
    "[SUCCESS]",
  ],
]
`;

exports[`calls fsx.outputFile 1`] = `
Array [
  Array [
    "wdir/yaml/single-symbol/single-output/output-filename/output.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/single-symbol/single-output/output-descriptor/output.schema.json",
    "{
	\\"symbol\\": \\"Foo\\",
	\\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/single-symbol/multiple-output/output-filename/abc.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/single-symbol/multiple-output/output-filename/def.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/single-symbol/multiple-output/output-filename/ghi.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/single-symbol/multiple-output/output-descriptor/non-descriptor.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/single-symbol/multiple-output/output-descriptor/default-descriptor.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/single-symbol/multiple-output/output-descriptor/custom-descriptor.schema.json",
    "{
	\\"symbol\\": \\"Foo\\",
	\\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/multiple-symbol/single-output/output-filename/foo.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/multiple-symbol/single-output/output-filename/bar.schema.json",
    "{
  \\"symbol\\": \\"Bar\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/multiple-symbol/single-output/output-descriptor/foo.schema.json",
    "{
	\\"symbol\\": \\"Foo\\",
	\\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/multiple-symbol/single-output/output-descriptor/bar.schema.json",
    "{
	\\"symbol\\": \\"Bar\\",
	\\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/multiple-symbol/multiple-output/output-filename/abc.foo.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/multiple-symbol/multiple-output/output-filename/def.foo.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/multiple-symbol/multiple-output/output-filename/ghi.foo.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/multiple-symbol/multiple-output/output-filename/abc.bar.schema.json",
    "{
  \\"symbol\\": \\"Bar\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/multiple-symbol/multiple-output/output-filename/def.bar.schema.json",
    "{
  \\"symbol\\": \\"Bar\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/multiple-symbol/multiple-output/output-filename/ghi.bar.schema.json",
    "{
  \\"symbol\\": \\"Bar\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/multiple-symbol/multiple-output/output-descriptor/foo.filename.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/multiple-symbol/multiple-output/output-descriptor/foo.default.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/multiple-symbol/multiple-output/output-descriptor/foo.custom.schema.json",
    "{
	\\"symbol\\": \\"Foo\\",
	\\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/yaml/multiple-symbol/multiple-output/output-descriptor/bar.4.schema.json",
    "{
    \\"symbol\\": \\"Bar\\",
    \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/single-symbol/single-output/output-filename/output.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/single-symbol/single-output/output-descriptor/output.schema.json",
    "{
	\\"symbol\\": \\"Foo\\",
	\\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/single-symbol/multiple-output/output-filename/abc.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/single-symbol/multiple-output/output-filename/def.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/single-symbol/multiple-output/output-filename/ghi.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/single-symbol/multiple-output/output-descriptor/non-descriptor.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/single-symbol/multiple-output/output-descriptor/default-descriptor.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/single-symbol/multiple-output/output-descriptor/custom-descriptor.schema.json",
    "{
	\\"symbol\\": \\"Foo\\",
	\\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/multiple-symbol/single-output/output-filename/foo.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/multiple-symbol/single-output/output-filename/bar.schema.json",
    "{
  \\"symbol\\": \\"Bar\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/multiple-symbol/single-output/output-descriptor/foo.schema.json",
    "{
	\\"symbol\\": \\"Foo\\",
	\\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/multiple-symbol/single-output/output-descriptor/bar.schema.json",
    "{
	\\"symbol\\": \\"Bar\\",
	\\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/multiple-symbol/multiple-output/output-filename/abc.foo.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/multiple-symbol/multiple-output/output-filename/def.foo.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/multiple-symbol/multiple-output/output-filename/ghi.foo.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/multiple-symbol/multiple-output/output-filename/abc.bar.schema.json",
    "{
  \\"symbol\\": \\"Bar\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/multiple-symbol/multiple-output/output-filename/def.bar.schema.json",
    "{
  \\"symbol\\": \\"Bar\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/multiple-symbol/multiple-output/output-filename/ghi.bar.schema.json",
    "{
  \\"symbol\\": \\"Bar\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/multiple-symbol/multiple-output/output-descriptor/foo.filename.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/multiple-symbol/multiple-output/output-descriptor/foo.default.schema.json",
    "{
  \\"symbol\\": \\"Foo\\",
  \\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/multiple-symbol/multiple-output/output-descriptor/foo.custom.schema.json",
    "{
	\\"symbol\\": \\"Foo\\",
	\\"type\\": \\"schema\\"
}
",
  ],
  Array [
    "wdir/json/multiple-symbol/multiple-output/output-descriptor/bar.4.schema.json",
    "{
    \\"symbol\\": \\"Bar\\",
    \\"type\\": \\"schema\\"
}
",
  ],
]
`;

exports[`messages 1`] = `
"
[SUCCESS]
"
`;

exports[`output contents 1`] = `
Array [
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/single-symbol/single-output/output-filename/output.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/single-symbol/single-output/output-descriptor/output.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/single-symbol/multiple-output/output-filename/abc.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/single-symbol/multiple-output/output-filename/def.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/single-symbol/multiple-output/output-filename/ghi.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/single-symbol/multiple-output/output-descriptor/non-descriptor.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/single-symbol/multiple-output/output-descriptor/default-descriptor.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/single-symbol/multiple-output/output-descriptor/custom-descriptor.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/multiple-symbol/single-output/output-filename/foo.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Bar",
      "type": "schema",
    },
    "filename": "wdir/yaml/multiple-symbol/single-output/output-filename/bar.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/multiple-symbol/single-output/output-descriptor/foo.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Bar",
      "type": "schema",
    },
    "filename": "wdir/yaml/multiple-symbol/single-output/output-descriptor/bar.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/multiple-symbol/multiple-output/output-filename/abc.foo.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/multiple-symbol/multiple-output/output-filename/def.foo.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/multiple-symbol/multiple-output/output-filename/ghi.foo.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Bar",
      "type": "schema",
    },
    "filename": "wdir/yaml/multiple-symbol/multiple-output/output-filename/abc.bar.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Bar",
      "type": "schema",
    },
    "filename": "wdir/yaml/multiple-symbol/multiple-output/output-filename/def.bar.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Bar",
      "type": "schema",
    },
    "filename": "wdir/yaml/multiple-symbol/multiple-output/output-filename/ghi.bar.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/multiple-symbol/multiple-output/output-descriptor/foo.filename.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/multiple-symbol/multiple-output/output-descriptor/foo.default.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/yaml/multiple-symbol/multiple-output/output-descriptor/foo.custom.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Bar",
      "type": "schema",
    },
    "filename": "wdir/yaml/multiple-symbol/multiple-output/output-descriptor/bar.4.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/single-symbol/single-output/output-filename/output.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/single-symbol/single-output/output-descriptor/output.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/single-symbol/multiple-output/output-filename/abc.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/single-symbol/multiple-output/output-filename/def.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/single-symbol/multiple-output/output-filename/ghi.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/single-symbol/multiple-output/output-descriptor/non-descriptor.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/single-symbol/multiple-output/output-descriptor/default-descriptor.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/single-symbol/multiple-output/output-descriptor/custom-descriptor.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/multiple-symbol/single-output/output-filename/foo.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Bar",
      "type": "schema",
    },
    "filename": "wdir/json/multiple-symbol/single-output/output-filename/bar.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/multiple-symbol/single-output/output-descriptor/foo.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Bar",
      "type": "schema",
    },
    "filename": "wdir/json/multiple-symbol/single-output/output-descriptor/bar.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/multiple-symbol/multiple-output/output-filename/abc.foo.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/multiple-symbol/multiple-output/output-filename/def.foo.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/multiple-symbol/multiple-output/output-filename/ghi.foo.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Bar",
      "type": "schema",
    },
    "filename": "wdir/json/multiple-symbol/multiple-output/output-filename/abc.bar.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Bar",
      "type": "schema",
    },
    "filename": "wdir/json/multiple-symbol/multiple-output/output-filename/def.bar.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Bar",
      "type": "schema",
    },
    "filename": "wdir/json/multiple-symbol/multiple-output/output-filename/ghi.bar.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/multiple-symbol/multiple-output/output-descriptor/foo.filename.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/multiple-symbol/multiple-output/output-descriptor/foo.default.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Foo",
      "type": "schema",
    },
    "filename": "wdir/json/multiple-symbol/multiple-output/output-descriptor/foo.custom.schema.json",
  },
  Object {
    "content": Object {
      "symbol": "Bar",
      "type": "schema",
    },
    "filename": "wdir/json/multiple-symbol/multiple-output/output-descriptor/bar.4.schema.json",
  },
]
`;
